htlcswitch/switch.go

```diff
@@ -2254,3 +2331,69 @@ func (s *Switch) FlushForwardingEvents() error {
 func (s *Switch) BestHeight() uint32 {
 	return atomic.LoadUint32(&s.bestHeight)
 }
+
+// evaluateDustThreshold takes in a ChannelLink, HTLC amount, and a boolean to
+// determine whether the default dust threshold has been exceeded. This
+// heuristic takes into account the trimmed-to-dust mechanism. The sum of the
+// commitment's dust with the mailbox's dust with the amount is checked against
+// the default threshold. If incoming is true, then the amount is not included
+// in the sum as it was already included in the commitment's dust. A boolean is
+// returned telling the caller whether the HTLC should be failed back.
+func (s *Switch) evaluateDustThreshold(link ChannelLink,
+	amount lnwire.MilliSatoshi, incoming bool) bool {
+
+	// Retrieve the link's current commitment feerate and dustClosure.
+	feeRate := link.getFeeRate()
+	isDust := link.getDustClosure()
+
+	// Evaluate if the HTLC is dust on either sides' commitment.
+	isLocalDust := isDust(feeRate, incoming, true, amount.ToSatoshis())
+	isRemoteDust := isDust(feeRate, incoming, false, amount.ToSatoshis())
+
+	if isLocalDust || isRemoteDust {
+		// Fetch the dust sums currently in the mailbox for this link.
+		cid := link.ChanID()
+		sid := link.ShortChanID()
+		mailbox := s.mailOrchestrator.GetOrCreateMailBox(cid, sid)
+		localMailDust, remoteMailDust := mailbox.DustPackets()
+
+		// If the htlc is dust on the local commitment, we'll obtain
+		// the dust sum for it.
+		if isLocalDust {
+			localSum := link.getDustSum(false)
+			localSum += localMailDust
+
+			// Optionally include the HTLC amount only for outgoing
+			// channels.
```

Comment:
"outgoing channels" or "outgoing htlcs" same above? Technically they're the same since the channel for an outgoing htlc is an outgoing channel, but I also sometimes think of an outgoing channel as one that our node opened.

Reply:
Will change to outgoing HTLC